lib/crystalball/graphql/map_generator/graphql_strategy/instance_definable_patch.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Crystalball
+  module GraphQL
+    module MapGenerator
+      class GraphQLStrategy
+        # Patches `GraphQL::Define::InstanceDefinable` to track object definitions
+        module InstanceDefinablePatch
+          class << self
+            # Patches `GraphQL::Define::InstanceDefinable.ensure_defined`.
+            def apply!
+              class << ::GraphQL::Define::InstanceDefinable
+                attr_accessor :executed
+              end
+              ::GraphQL::Define::InstanceDefinable::ClassMethods.class_eval do
+                prepend InstanceDefinablePatch
+              end
+            end
+          end
+
+          # Will replace original `GraphQL::Define::InstanceDefinable.ensure_defined`.
+          # Marks `executed` as true and then calls the original `define`
+          def ensure_defined(*)
+            ::GraphQL::Define::InstanceDefinable.executed = true
+            super
+          end
```

Comment:
why do we have to pollute GraphQL module with our flag accessor?
where is revert functionality?

Reply:
It also doesn't save the type that was defined. There might be multiple schemas defined and they'll have separate definition trees.